sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParserUtils.scala

```diff
@@ -178,4 +179,11 @@ object StaxXmlParserUtils {
       }
     }
   }
+
+  def consumeNextEndElement(parser: XMLEventReader): Unit = {
+    parser.nextEvent() match {
+      case _: EndElement => // do nothing
+      case _ => throw new IllegalStateException("Invalid state")
+    }
```

Comment:
Extra events (comments, etc.) before EndElement need to be skipped.

Reply:
I think comments are removed before it reach parser. WDYT?

Reply:
You are right. We are filtering comments [here](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParserUtils.scala#L43). Do we need special handling for CDATA or EntityReferences?

Reply:
Discussed offline. The current implementation should work fine with `CDATA` and we will filter out `EntityReferences`